python/py-avro/Portfile

```diff
@@ -18,17 +19,36 @@ long_description    ${description}
 
 homepage            https://avro.apache.org/
 
-checksums           rmd160  61c45d1cfa9914306401cb88d880ac63602d43cc \
-                    sha256  f456539183df0d6912daedfbf0f94d7cf69b4fba7200490e8a06d1e8f23af43d \
-                    size    1845163
+checksums           rmd160  e398cc448a227de8754eb60848eb4a47f0633265 \
+                    sha256  4e94b43ebdb1081fd842644bda2caaddc586c1b608efe9236d348d01bf0866a4 \
+                    size    1899682
 
 python.versions     37 38
 
 if {${name} ne ${subport}} {
     depends_build-append \
-                        port:py${python.version}-setuptools
-
-    worksrcdir          ${distname}/lang/py3
+                    port:py${python.version}-setuptools
+
+    depends_test-append \
+                    port:py${python.version}-pytest
+
+    build.dir       ${worksrcpath}/lang/py3
+    destroot.dir    ${worksrcpath}/lang/py3
+    test.dir        ${worksrcpath}/lang/py3
+    test.run        yes
+    test.cmd        py.test-${python.branch}
+    test.target
+    test.env-append \
+                    PYTHONPATH=${worksrcpath}/lang/py3/build/lib
+
+    post-destroot {
+        set docdir ${prefix}/share/doc/${subport}
+        xinstall -d ${destroot}${docdir}
+        xinstall -m 0644 -W ${worksrcpath} LICENSE.txt README.md \
+            ${destroot}${docdir}
+    }
 
     livecheck.type      none
+} else {
+    livecheck.type      pypi
```

Comment:
you're using the `github` PG so please use the livecheck from GitHub as well. If you don't want pre-release versions set `github.livecheckregex` properly.

Reply:
Two points:
* pypi's automated version checking is *much* more reliable than many GitHub authors/repos in providing version tags, so it makes sense to me to use pypi for live checks even though the download source is from GitHub.
* If it's absolutely necessary to use GitHub version checking, please post the exact `Portfile` code needed.

Reply:
you can easily find how to do this in other ports, but okay here it is: `github.livecheck.regex {([0-9.]+)}`